xarray_subset_grid/grids/ugrid.py

```diff
@@ -153,15 +153,19 @@ def subset_polygon(
 
         # NOTE: UGRIDS can be zero-indexed OR one-indexed!
         #       see the UGRID spec.
-        tris = face_node_connectivity - face_node_start_index
-        tri_mask = node_inside[tris]
+        # NOTE: The face_node_connectivity may contain masked elements as
+        #       per the UGRID spec. We set these to -1 and only slice the
+        #       valid elements.
+        tris = face_node_connectivity.fillna(-1) - face_node_start_index
```

Comment:
`fillna` does nothing if there no NaN masked values, so this is harmless if not applicable